lib/ejs.js

```diff
@@ -721,75 +712,97 @@ Template.prototype = {
+    if (line[0] == '<') {
+      isHandled = true;
+      switch (line) {
+      case '<' + d:
+      case '<' + d + '_':
+        this.mode = Template.modes.EVAL;
+        break;
+      case '<' + d + '=':
+        this.mode = Template.modes.ESCAPED;
+        break;
+      case '<' + d + '-':
+        this.mode = Template.modes.RAW;
+        break;
+      case '<' + d + '#':
+        this.mode = Template.modes.COMMENT;
+        break;
+      case '<' + d + d:
+        this.mode = Template.modes.LITERAL;
+        this.source += '    ; __append("' + line.replace('<' + d + d, '<' + d) + '")' + '\n';
+        break;
+      default:
+        isHandled = false;
       }
-
-      this.mode = null;
-      this.truncate = line.indexOf('-') === 0 || line.indexOf('_') === 0;
-      break;
-    default:
-        // In script mode, depends on type of tag
-      if (this.mode) {
-          // If '//' is found without a line break, add a line break.
-        switch (this.mode) {
-        case Template.modes.EVAL:
-        case Template.modes.ESCAPED:
-        case Template.modes.RAW:
-          if (line.lastIndexOf('//') > line.lastIndexOf('\n')) {
-            line += '\n';
-          }
-        }
-        switch (this.mode) {
-            // Just executing code
-        case Template.modes.EVAL:
-          this.source += '    ; ' + line + '\n';
-          break;
-            // Exec, esc, and output
-        case Template.modes.ESCAPED:
-          this.source += '    ; __append(escapeFn(' + stripSemi(line) + '))' + '\n';
-          break;
-            // Exec and output
-        case Template.modes.RAW:
-          this.source += '    ; __append(' + stripSemi(line) + ')' + '\n';
-          break;
-        case Template.modes.COMMENT:
-              // Do nothing
+      if (isHandled && this.mode != Template.modes.LITERAL) {
+        if (this.isInTag)
+          throw new Error('Could not find matching close tag for "' + this.isInTag + '".');
+        this.isInTag = line;
+      }
+    }
+    if (! isHandled) {
+      if (line[line.length-1] == '>') {
+        isHandled = true;
+        switch (line) {
+        case d + d + '>':
+          this.mode = Template.modes.LITERAL;
+          this.source += '    ; __append("' + line.replace(d + d + '>', d + '>') + '")' + '\n';
           break;
-            // Literal <%% mode, append as raw output
-        case Template.modes.LITERAL:
-          this._addOutput(line);
+        case d + '>':
+        case '-' + d + '>':
+        case '_' + d + '>':
+          this.isInTag = undefined;
+          if (this.mode == Template.modes.LITERAL) {
+            this._addOutput(line);
+          }
+    
+          this.mode = null;
+          this.truncate = line.indexOf('-') === 0 || line.indexOf('_') === 0;
           break;
+        default:
+          isHandled = false;
         }
       }
-        // In string mode, just add the output
-      else {
-        this._addOutput(line);
+      if (! isHandled) {
```

Comment:
Space between `!` "not" operator and variable name is not needed. Should be simply `!isHandled`.